aws/graph-explorer
# (bugfix/issue-55) Support non-string based ids for gremlin connector

Issue #55

#### Description of changes:
- Updates Gremlin connector to take into account non-string based ids for nodes and edges during the mapping

#### How Has This Been Tested?
- Mount a local instance of Tinkerpop `docker run --rm -p 8182:8182 tinkerpop/gremlin-server conf/gremlin-server-rest-modern.yaml` using the REST configuration. The default graph contains vertices of type person and software whose ids are like `{"@type":"g:Int64","@value":1}`. These kind of non-string ids are the reason of the issue.
- Create a new configuration with the following parameters:
  - Public endpoint: http://localhost:8182
  - Enable cache: false (only to guarantee that all the requests are always performed against the database)

- After synchronization, inspect the 4 people of the 2 software in the connection details
- "Node id" column now is not `[Object object]` but it shows the numeric value of the id.
- It has also been checked that this change does not affect to search or expand capabilities.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
